trunner/harness/base.py

```diff
@@ -92,9 +92,14 @@ def _reboot_hard(self):
         self.host.set_power(True)
         time.sleep(0.5)
 
-    def _reboot_dut_gpio(self, hard):
+    def _reboot_via_jtag(self, ftdi):
+        pass
+
+    def _reboot_dut_gpio(self, hard, ftdi=False):
```

Comment:
this is generic `Rebooter` class used for all targets. Please don't use `ftdi` term here.

Maybe we should describe (as docstrings) which methods of rebooting we do support and what is needed for them to work.

The third generic option to reboot you're adding is "rebooting by debugger" (regardless of the debug protocol (jtag/swd) or debugger device (jlink/mcu-link/ftdi-based/lauterbach, etc.))